### PR TITLE
Fire rate indepency blah blah blah 2: electric boogalo

### PR DESCRIPTION
**Pull request by Qustinnus**
🆑 Qustinnus / Floyd
tweak: Makes weapons fire indepently from each other, instead of all firing at once
/🆑

because i cant into git.

### CODE REVIEW

File: code/datums/ship.dm [hunk 1]
```diff
@@ -268,27 +267,154 @@ var/next_ship_id
...
+	attack_data = /datum/ship_attack/flame_bomb
+
+
+			//Stun Bombs
+/datum/component/weapon/stunbomb
+	name = "stunbomber"
+	cname = "stunbomber"
+	fire_rate = 300
+
+	attack_data = /datum/ship_attack/stunbomb
```
Comment (Super3222) [inline code/datums/ship.dm]: `/datum/ship_attack/stun_bomb`

Comment (Qustinnus) [inline code/datums/ship.dm]: yeah I saw. I'm currently doing a commit it'll be up in about 30 minutes with a lot of extra stuff. Thanks for pointing it out, though :)

File: code/controllers/subsystem/ships.dm [hunk 2]
```diff
@@ -78,9 +78,9 @@ var/global/list/ftl_weapons_consoles = list()
 			return faction_A.relations[i]
 
 
-/datum/subsystem/ship/proc/calculate_damage_effects(var/datum/starship/S)
-
-	S.fire_rate = round(initial(S.fire_rate) * factor_damage_inverse(SHIP_WEAPONS,S))
+/datum/subsystem/ship/proc/calculate_damage_effects(var/datum/starship/S, var/datum/component/weapon/W)
+	for(W in S)
+		W.fire_rate = round(initial(W.fire_rate) * factor_damage_inverse(SHIP_WEAPONS,S))
```
Comment (ninjanomnom) [inline code/controllers/subsystem/ships.dm]: You take an input weapon then use it in a for loop for ships, wtf

should be for(var/datum/component/weapon/W in S.components)

and take out the input var/datum/component/weapon/W

**Comment by ninjanomnom** [comment]
The ship combat code in it's subsystem disgusts me, so much copypasta

File: code/controllers/subsystem/ships.dm (hunk 2 above)
Comment (Qustinnus) [inline code/controllers/subsystem/ships.dm]: do i look like a good coder to you u banana. ill change it.

**Comment by Vivalas** [comment]
@ninjanomnom Which parts exactly?

File: code/controllers/subsystem/ships.dm [hunk 3]
```diff
@@ -79,8 +79,8 @@ var/global/list/ftl_weapons_consoles = list()
 
 
 /datum/subsystem/ship/proc/calculate_damage_effects(var/datum/starship/S)
-
-	S.fire_rate = round(initial(S.fire_rate) * factor_damage_inverse(SHIP_WEAPONS,S))
+	for(var/datum/component/weapon/W in S)
```
Comment (ninjanomnom) [inline code/controllers/subsystem/ships.dm]: S.components

**Pull request merged**